apps/examples/load_fw/zynqmp_r5_lcm_rproc_example.c

```diff
@@ -137,86 +137,79 @@ void r5_rproc_remove(struct remoteproc *rproc)
 	}
 }
 
-void *r5_rproc_mmap(struct remoteproc *rproc,
-		    metal_phys_addr_t *pa, metal_phys_addr_t *da,
-		    size_t size, unsigned int attribute,
-		    struct metal_io_region **io)
+int r5_rproc_mmap(struct remoteproc *rproc,
+		  metal_phys_addr_t *pa, metal_phys_addr_t *da,
+		  void **va, size_t size, unsigned int attribute,
+		  struct metal_io_region **io)
 {
 	struct remoteproc_mem *mem;
+	struct metal_io_region *tmpio;
 	struct r5_rproc_priv *priv;
-	metal_phys_addr_t lpa, lda;
 
 	priv = rproc->priv;
 
-	if (!da || !pa)
-		return NULL;
 	LPRINTF("%s: pa=0x%x, da=0x%x, size=0x%x, atrribute=0x%x\r\n",
 		__func__, *pa, *da, size, attribute);
-	lda = *da;
-	lpa = *pa;
 	if (!attribute)
 		attribute = NORM_SHARED_NCACHE | PRIV_RW_USER_RW;
-	if (lda <= 0x40000) {
+	if (*da <= 0x40000) {
```

Comment:
check if da is invalid pointer before checking contents

Reply:
don't check null pointer here since mmap is only called by remoteproc_mmap and this function already ensure va/pa/da never become null.